.github/workflows/test-sui.yaml

```diff
@@ -0,0 +1,144 @@
+        uses: ./.github/actions/setup-sui
+        with:
+          SUI_VERSION: ${{ env.SUI_VERSION }}
+
+      - name: Spin up Sui Network
+        run: nohup sh -c "sui-test-validator" > nohup.out 2> nohup.err < /dev/null &
+
+      - name: Sleep for 30 seconds
+        run: sleep 30s
+        shell: bash
+
+      - name: Prepare local.json
+        run: |
+          echo '{
+                "sui": {
+                    "name": "Sui",
+                    "axelarId": "sui",
+                    "networkType": "localnet",
+                    "tokenSymbol": "SUI",
+                    "rpc": "http://127.0.0.1:9000",
+                    "faucetUrl": "http://127.0.0.1:9123",
+                    "contracts": {
+                      "AxelarGateway": {}
+                    }
+                }
+            }' > ./axelar-chains-config/info/local.json
+
+      # Create .env file with default hardhat private key that's prefunded
+      - name: Prepare .env
+        run: |
+          echo 'PRIVATE_KEY=suiprivkey1qyw6c8t5ws65fcvlcw5rhyt2jd8d6ad6e526shgktlm0x3aelq9rx2lp4wq' >> .env
+          echo 'ENV=local' >> .env
+          echo 'SKIP_EXISTING = true' >> .env
+
+      - name: Display local.json
+        run: cat ./axelar-chains-config/info/local.json
+
+      - name: Request SUI from faucet
+        run: node sui/faucet.js
+
+      ###### Command: Deploy Contract ######
+
+      - name: Deploy AxelarGateway
+        run: node sui/deploy-contract deploy AxelarGateway --signers wallet
+
+      - name: Deploy GasService
+        run: node sui/deploy-contract deploy GasService
+
+      - name: Deploy Operators
+        run: node sui/deploy-contract deploy Operators
+
+      - name: Deploy Test
+        run: node sui/deploy-contract deploy Test
+
+      ###### Command: Gas Service ######
+
+      - name: Pay Gas
+        run: node sui/gas-service.js payGas --amount 100 ethereum 0x6f24A47Fc8AE5441Eb47EFfC3665e70e69Ac3F05 0xba76c6980428A0b10CFC5d8ccb61949677A61233 0x1234
+
+      - name: Refund Gas
+        run: node sui/gas-service.js refund 0x2 --amount 1
+
+      - name: Collect Gas
+        run: node sui/gas-service.js collectGas --amount 0.1
+
+      ###### Command: GMP ######
+
+      - name: Execute Outgoing Call Contract
+        run: node sui/gmp.js sendCall ethereum 0x6f24A47Fc8AE5441Eb47EFfC3665e70e69Ac3F05 0.1 0x1234
+
+      # TODO: Fix error in this command
+      - name: Execute Incoming Call Contract
+        continue-on-error: true
+        run: |
+          node sui/gateway.js approve --proof wallet ethereum 0x32034b47cb29d162d9d803cc405356f4ac0ec07fe847ace431385fe8acf3e6e5-2 0x4F4495243837681061C4743b74B3eEdf548D56A5 0x6ce0d81b412abca2770eddb1549c9fcff721889c3aab1203dc93866db22ecc4b 0x56570de287d73cd1cb6092bb8fdee6173974955fdef345ae579ee9f475ea7432
+          node sui/gmp.js execute ethereum 0x32034b47cb29d162d9d803cc405356f4ac0ec07fe847ace431385fe8acf3e6e5-2 0x4F4495243837681061C4743b74B3eEdf548D56A5 0x1234
```

Comment:
@milapsheth i saw your message in `smart-contracts` channel and looks like we're going to fix this script? In that case, we can wait until the fix is completed.

Reply:
yep, merged my PR with the fixes